main/test/ca/uwaterloo/flix/language/phase/TestRedundancy.scala

```diff
@@ -451,6 +465,17 @@ class TestRedundancy extends FunSuite with TestUtils {
     expectError[RedundancyError.UnusedFormalParam](result)
   }
 
+  test("UnusedFormalParam.NewObject.01") {
+    val input =
+      """
+        |def f(): ##java.lang.Comparable & Impure =
+        |   object ##java.lang.Comparable {
+        |     def compareTo(x: ##java.lang.Object, _y: ##java.lang.Object): Int32 =
+        |       0
+        |   }
+        """.stripMargin
```

Comment:
Don't forget to compile and expectError :)

Reply:
D'oh!